src/util/FileStorage.php

```diff
@@ -0,0 +1,35 @@
+<?php
+namespace LogHero\Client;
+
+
+class FileStorage implements StorageInterface {
+    private $storageFilename;
+
+    public function __construct($storageFilename) {
+        static::verifyWriteAccess($storageFilename);
+        $this->storageFilename = $storageFilename;
+    }
+
+    public function set($jsonDataAsString) {
+        file_put_contents($this->storageFilename, $jsonDataAsString);
+        chmod($this->storageFilename, 0666);
+    }
+
+    public function get() {
+        if (file_exists($this->storageFilename)) {
+            return file_get_contents($this->storageFilename);
+        }
+        return null;
+    }
+
+    public static function verifyWriteAccess($fileLocation) {
```

Comment:
There are some places using files. Maybe a discrete "File" class would be a good way to go. This class could verify the write access and encapsulate basic things like changing the file mode, checking for file existance and simple get contents.

Reply:
That's the idea of this class. However, in most cases I need to implement custom handling of missing files, some files need locking etc. I am not sure if a generic interface against file access is easy possible.